Hero/Hero.swift

```diff
@@ -338,22 +339,19 @@ extension Hero: UIViewControllerAnimatedTransitioning {
 
 extension Hero:UIViewControllerTransitioningDelegate {
   fileprivate var interactiveTransitioning: UIViewControllerInteractiveTransitioning? {
-    if let fvc = fromViewController as? HeroViewControllerDelegate {
-      interactive = fvc.wantInteractiveHeroTransition?() ?? false
-    }
-    if !interactive, let navigationController = fromViewController as? UINavigationController,
-      let fvc = navigationController.topViewController as? HeroViewControllerDelegate {
-      interactive = fvc.wantInteractiveHeroTransition?() ?? false
+    closureProcessForHeroDelegate(vc: fromViewController!) {
+      interactive = $0.wantInteractiveHeroTransition?() ?? false
     }
+    
     if !interactive {
-      plugins = Hero.enabledPlugins.map({ return $0.init() })
-      for plugin in plugins{
-        if plugin.wantInteractiveHeroTransition(){
+      Hero.enabledPlugins.map { $0.init() }.forEach {
```

Comment:
Here it makes sense. I can understand the justification of using forEach here when chaining.